### ISSUE TITLE
Line with more than 32768 points doesn't display correctly using "line_dash" option

### ISSUE THREAD

**Issue by cgmorton** [bokeh/bokeh#2841]
When I plot an array with more than 32768 (2**15) values and I use the "line_dash" option, the line appears to stops at ~32768 values and starts over at the initial x value.  I didn't see anything similar when I searched the issues, but sorry in advance if I missed it.

Code to reproduce:
```
import datetime
from bokeh.plotting import figure, output_file, show
from bokeh.models import Range1d
import numpy as np

dt_array = np.arange('1950-01-01', '2099-12-31', dtype='datetime64[D]')

output_file('line_dash_error.html')
f = figure(x_axis_type='datetime')

f.line(dt_array, 0.8 * np.ones(dt_array.shape), line_width=3)
f.line(dt_array, 0.6 * np.ones(dt_array.shape), line_dash="dotted", line_width=3)
f.line(dt_array, 0.4 * np.ones(dt_array.shape), line_dash="dashdot", line_width=3)
f.line(dt_array, 0.2 * np.ones(dt_array.shape), line_dash=[4,8], line_width=3)

f.set(y_range=Range1d(0, 1))
show(f)
```

Screenshot of what I see:
![line_dash_error](https://cloud.githubusercontent.com/assets/9002566/9820779/e6314292-586a-11e5-95e0-ccaa92e47911.png)

**Comment by bryevdv** [comment]
For something like this we will need information about versions, browsers, etc.

**Comment by cgmorton** [comment]
Sorry!  Windows 7 64-bit, Python 2.7.8 64-bit, Bokeh 0.9.3 (latest version using Conda), Chrome version 45.0.2454.85 and Firefox 40.0.3.

**Comment by bryevdv** [comment]
I can reproduce this on FF but not on the version of Chrome I have. Unfortunately this is indicative of an upstream issue with the HTML Canvas implementation. I'm not sure there's alot we can do.

**Comment by bryevdv** [comment]
This is fixed in recent FF (`48.0b3`) As I suggested this is an upstream issue, and will be resolved when browsers catch up (if they have not all done so already) Another option to try if you are pinned to old browsers, or still see the issue in some browsers is to use `webgl=True` now that webgl supports lines

<img width="715" alt="screen shot 2016-07-02 at 11 17 37 pm" src="https://cloud.githubusercontent.com/assets/1078448/16543738/34a0c8a2-40ab-11e6-8bbc-34c974c16de7.png">
